torch/_subclasses/fake_tensor.py

```diff
@@ -128,6 +129,16 @@ class MetadataMismatchError(RuntimeError):
     reason: str
 
 
+class FakeTensorTLS(threading.local):
+    allow_non_fake_inputs_override: bool
+
+    def __init__(self) -> None:
+        self.allow_non_fake_inputs_override = False
```

Comment:
maybe a comment that this can be used to override allow_non_fake_inputs=False, but not allow_non_fake_inputs=True? I got a bit confused reading this

Alternatively, we could set this to be None as the default and when it is ever non-none then we pull from it instead of allow_non_fake_inputs.

Reply:
Ah good call.

> Alternatively, we could set this to be None as the default and when it is ever non-none then we pull from it instead of allow_non_fake_inputs.

Yeah this is probably more intuitive. I'll update.